server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/impl/EventSourcingFilteringManagement.java

```diff
@@ -56,8 +58,12 @@ public EventSourcingFilteringManagement(EventStore eventStore) {
     }
 
     @Override
-    public Publisher<Void> defineRulesForUser(Username username, List<Rule> rules) {
-        return Mono.from(eventSourcingSystem.dispatch(new DefineRulesCommand(username, rules))).then();
+    public Publisher<Version> defineRulesForUser(Username username, List<Rule> rules, Optional<Version> ifInState) {
+        return Mono.from(eventSourcingSystem.dispatch(new DefineRulesCommand(username, rules, ifInState)))
+            .then(Mono.from(eventStore.getEventsOfAggregate(new FilteringAggregateId(username)))
+                .map(History::getVersionAsJava)
+                .map(optionalVersion -> optionalVersion.map(eventId -> new Version(eventId.value()))
+                    .orElse(null)));
```

Comment:
1. Is that really a `null` value?

In which case is it better than an Optional?

Are we begging for bugs?

Also, shouldn't we have the `EventId -> Version` conversion centralized in a single place? (this would avoid divergences.)